### PR TITLE
Icon of floating window

### PR DESCRIPTION
**Pull request by mvidelgauz**
Currently, the floating window title reflects the title of the active dock widget but doesn't reflect its icon.
This PR fills the gap.

### CODE REVIEW

File: src/FloatingDockContainer.cpp [hunk 1]
```diff
@@ -537,8 +537,9 @@ void CFloatingDockContainer::onDockAreasAddedOrRemoved()
 	if (TopLevelDockArea)
 	{
 		d->SingleDockArea = TopLevelDockArea;
-		d->setWindowTitle(
-		    d->SingleDockArea->currentDockWidget()->windowTitle());
+		CDockWidget* CurrentWidget = d->SingleDockArea->currentDockWidget();
+		d->setWindowTitle(CurrentWidget->windowTitle());
+		setWindowIcon(CurrentWidget->icon());
```
Comment (githubuser0xFFFF) [inline src/FloatingDockContainer.cpp]: Set the window icon only, if the CurrentWidget->icon() is valid - if the CurrentWidget has no icon, then set the Application icon

File: src/FloatingDockContainer.cpp [hunk 2]
```diff
@@ -560,19 +562,24 @@ void CFloatingDockContainer::updateWindowTitle()
 	auto TopLevelDockArea = d->DockContainer->topLevelDockArea();
 	if (TopLevelDockArea)
 	{
-		d->setWindowTitle(TopLevelDockArea->currentDockWidget()->windowTitle());
+		CDockWidget* CurrentWidget = TopLevelDockArea->currentDockWidget();
+		d->setWindowTitle(CurrentWidget->windowTitle());
+		setWindowIcon(CurrentWidget->icon());
```
Comment (githubuser0xFFFF) [inline src/FloatingDockContainer.cpp]: Set window icon only if the dock widget icon returns a valid icon

**Comment by rolivav** [comment]
I agree with @githubuser0xFFFF , an option would be nice. It's funny, in my application it would be useful to have the same title in floating windows as the main application, or even a custom floating name, since I have tabs always shown.

**Comment by mvidelgauz** [comment]
` in my application it would be useful to have the same title in floating windows as the main application`
while I am working on `DynamicFloatingWidgetIcon` (or any other good name) in these parts of code I can do this also, if @githubuser0xFFFF will permit :) (then the duty of invention of yet another good flag name will be on you @rolivav)

**Comment by rolivav** [comment]
Haha so many flags, it is starting to become difficult to name them. Could be something like `UseApplicationNameInFloatingWindows`, what do you think? Thanks for offering to implement this btw, much appreciated.

**Comment by mvidelgauz** [comment]
@githubuser0xFFFF what about:
`FloatingContainerHasWidgetIcon` - false by default and true for my application
`FloatingContainerHasWidgetTitle` - true by default and false for @rolivav's application

File: src/FloatingDockContainer.cpp (hunk 1 above)
Comment (mvidelgauz) [reply src/FloatingDockContainer.cpp]: Fixed

File: src/FloatingDockContainer.cpp (hunk 2 above)
Comment (mvidelgauz) [reply src/FloatingDockContainer.cpp]: Fixed

**Comment by mvidelgauz** [comment]
@githubuser0xFFFF  Added those two new flags, could you please review the current change?

**Comment by githubuser0xFFFF** [comment]
I like the names `FloatingContainerHasWidgetIcon` and `FloatingContainerHasWidgetTitle`. I just merged your pull request - thank you. I think, we need to write a tutorial or getting started to show how to use all these flags 😉

**Comment by mvidelgauz** [comment]
Thank you @githubuser0xFFFF 
@rolivav Enjoy new flag :)

**Pull request merged**